File: geoportal/src/com/esri/gpt/server/csw/client/CswProfiles.java
/* See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * Esri Inc. licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package com.esri.gpt.server.csw.client;

import java.io.IOException;
import java.net.URL;
import java.util.Properties;

import javax.xml.parsers.DocumentBuilder;
import javax.xml.parsers.DocumentBuilderFactory;
import javax.xml.parsers.ParserConfigurationException;
import javax.xml.xpath.XPath;
import javax.xml.xpath.XPathExpressionException;
import javax.xml.xpath.XPathFactory;

import org.w3c.dom.Document;
import org.w3c.dom.Node;
import org.w3c.dom.NodeList;
import org.xml.sax.InputSource;
import org.xml.sax.SAXException;

import com.esri.gpt.framework.search.SearchXslProfiles;
import com.esri.gpt.framework.util.ResourcePath;
import com.esri.gpt.framework.util.Val;


/**
 * The collection of CSw profiles.
 * 
 */
public class CswProfiles extends SearchXslProfiles<CswProfile> {

// instance variables ==========================================================

/** The configuration_folder_path. */
private String  configuration_folder_path = ""; 

/** The profiles file **/
private static final String CSW_PROFILES_FILE = "CSWProfiles.xml";

// constructors ================================================================
/**
 * Instantiates a new csw profiles.
 */
public CswProfiles() {
  super();
 
}

// properties ==================================================================


// methods =====================================================================
/**
 * Loads the profile details from configuration file.
 * 
 * The profiles details are loaded in the collection. Duplicate or invalid
 * profiles are ignored.
 * 
 * @param filename
 * @throws ParserConfigurationException 
 * @throws SAXException 
 * @throws ParserConfigurationException 
 * @throws IOException 
 * @throws SAXException 
 * @throws XPathExpressionException 
 */
public void loadProfilefromConfig(String filename)
    throws ParserConfigurationException, SAXException, IOException,
    XPathExpressionException {
  
  configuration_folder_path = this.getConfigurationFolderPath();
  if(Val.chkStr(filename).equals("")) {
    filename = CSW_PROFILES_FILE;
  }
  super.loadProfilefromConfig(filename, new CswProfile(), "Profile");
 
}

/**
 * Load profile from default profiles config file.
 * 
 * @throws XPathExpressionException the x path expression exception
 * @throws ParserConfigurationException the parser configuration exception
 * @throws SAXException the SAX exception
 * @throws IOException Signals that an I/O exception has occurred.
 */
public void loadProfilefromConfig() throws XPathExpressionException,
    ParserConfigurationException, SAXException, IOException {
  loadProfilefromConfig(null);
}

}

File: components/desktop/CswClient/branches/10.1/common/CswProfiles.cs
/* See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * Esri Inc. licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
using System;
using System.Collections.Generic;
using System.Text;
using System.Xml;

namespace com.esri.gpt.csw {
    /// <summary>
    /// The collection of CSw profiles
    /// </summary>
   public class CswProfiles : CswObjects
   {       
       public static String DEFAULT_CSW_NAMESPACE = "http://www.opengis.net/cat/csw";
       #region Propeties
       public new CswProfile this[string key] {
            get { return (CswProfile)base[key]; }
        }
       public new CswProfile this[int index] {
            get {
                return (CswProfile)base[index];
            }
        }
       #endregion
       #region "PublicMethods"

        /// <summary>
        /// Add a key value pair to profile collection. 
        /// </summary>
        /// <remarks>
        /// Add to profile collection.
        /// </remarks>
        /// <param name="param1">The key which is the url hashcode for the profile</param>
        /// <param name="param2">The profile object</param>

        public void AddProfile(object key, CswProfile profile) {
            base.Add(key,profile);
        }

        /// <summary>
        /// Loads the profile details from configuration file. 
        /// </summary>
        /// <remarks>
        /// The profiles details are loaded in the collection.
        /// Duplicate or invalid profiles are ignored.
        /// </remarks>
        /// <param name="param2">the profile configuration file</param>
        public void loadProfilefromConfig(string filename) {
            try
            {
                XmlDocument doc = new XmlDocument();
                doc.Load(filename);
                XmlNodeList xmlnodes = doc.GetElementsByTagName("Profile");

                foreach (XmlNode xmlnode in xmlnodes) {
                    string cswnamespace = "";
                    try {
                        cswnamespace = xmlnode.SelectSingleNode("CswNamespace").InnerText.Trim();
                    }
                    catch (NullReferenceException ex) {
                        cswnamespace = CswProfiles.DEFAULT_CSW_NAMESPACE;
                    }
                    if (cswnamespace.Length <= 0) {
                        cswnamespace =CswProfiles.DEFAULT_CSW_NAMESPACE ;
                    }
                    string id = xmlnode.SelectSingleNode("ID").InnerText.Trim();
                    string name = xmlnode.SelectSingleNode("Name").InnerText.Trim();                    
                    string description = xmlnode.SelectSingleNode("Description").InnerText.Trim();
                    string requestxslt = xmlnode.SelectSingleNode("GetRecords/XSLTransformations/Request").InnerText.Trim();
                    string responsexslt = xmlnode.SelectSingleNode("GetRecords/XSLTransformations/Response").InnerText.Trim();                   
                    string requestKVPs = xmlnode.SelectSingleNode("GetRecordByID/RequestKVPs").InnerText.Trim();
                    bool livedatamap = bool.Parse(xmlnode.SelectSingleNode("SupportContentTypeQuery").InnerText.Trim());
                    bool extentsearch = bool.Parse(xmlnode.SelectSingleNode("SupportSpatialQuery").InnerText.Trim());
                    bool spatialboundary = bool.Parse(xmlnode.SelectSingleNode("SupportSpatialBoundary").InnerText.Trim());
                    string metadataxslt = "";
                    XmlNode node = xmlnode.SelectSingleNode("GetRecordByID/XSLTransformations/Response");
                    if (node != null) { metadataxslt = node.InnerText.Trim(); }

                    string displayResponseXslt = "";
                    node = xmlnode.SelectSingleNode("GetRecordByID/XSLTransformations/StyleResponse");
                    if (node != null) { displayResponseXslt = node.InnerText.Trim(); }

                    // generate full path for XSLT files
                    // handles both absolute path and relative path, assuming profile xml is the root for relative path
                    string profileDirName = System.IO.Path.GetDirectoryName (filename);
                    requestxslt = System.IO.Path.Combine(profileDirName, requestxslt);
                    responsexslt = System.IO.Path.Combine(profileDirName, responsexslt);
                    if (metadataxslt != "") { metadataxslt = System.IO.Path.Combine(profileDirName, metadataxslt); }

                    if (displayResponseXslt != "") { displayResponseXslt = System.IO.Path.Combine(profileDirName, displayResponseXslt); }

                    this.AddProfile(id, new CswProfile(id, name, cswnamespace, description, requestKVPs, requestxslt, responsexslt, metadataxslt, displayResponseXslt,livedatamap, extentsearch, spatialboundary));
                }

            }
            catch (Exception ex)
            {
               Console.WriteLine(ex.Message);
                throw ex;
            }

        }

        #endregion
    }
}
